### DISCUSSION TITLE
CatBoost Roadmap: January-March 2023

### DISCUSSION THREAD

**Discussion by kizill** [catboost/catboost#2279] [General]
Our main goals for January-March 2023 are solving long lasting problems with CatBoost binaries build and to make CatBoost with popular tools, like Optuna and others.

###  Building CatBoost
- Builds for Linux & MacOS should be more open and understandable for all the developers, so we are switching to CMake + Conan +ninja for building and releasing artifacts
- Python wheels now build with `mk_wheel.py`, we'd like to make `setup.py` and use it for building, packing and releasing wheels for Linux & MacOS

### External tools and CatBoost
- Better support for parameters tuning: integration with hyperopt & optuna, make more examples
- Support for experiment tracking: better callbacks for mlflow, W&B support

### Dask support
We'd like to make some steps to make distributed CatBoost training easier and make minimally working version in Dask environment.

**Comment by celestinoxp** [comment]
Hi,
>Builds for Linux & MacOS should be more open...

you talk about Linux and MacOS... I ask and Windows too, right? :)

I think the next few months are going to be very interesting! I'm with you :) Thank you

**Reply by kizill** [reply]
Yes, we are still fixing Windows build problems. We are trying to fully switch to CMake or to MSVS solution files & msbuild. We can't fully migrate to clang toolkit on windows as we have CUDA code and it's compilation tightly bound with MSVC toolkits.

**Comment by makart19** [comment]
Hi,

As part of your migration to CMake can you please add support of:
-	building without CUDA (smth like  ‘-DHAVE_CUDA=no’ as it was for ‘ya make’).
-	building for riscv64 CPU.

Thanks in advance!

**Reply by andrey-khropov** [reply]
> Hi,
> 
> As part of your migration to CMake can you please add support of:
> 
>     * building without CUDA (smth like  ‘-DHAVE_CUDA=no’ as it was for ‘ya make’).

Yes, there's an issue about that: #2298 

> 
>     * building for riscv64 CPU.

Please create an issue about that. But I'm not sure that it will be supported soon.

> 
> 
> Thanks in advance!

**Reply by makart19** [reply]
Thank you for quick response, here it is https://github.com/catboost/catboost/issues/2314.

**Comment by kesav02** [comment]
Hello, are you planing to extend the C languge api to include training model functionality?
